source/common/http/http1/parser_impl.cc

```diff
@@ -0,0 +1,225 @@
+#include "common/http/http1/parser_impl.h"
+
+#include <llhttp.h>
+
+#include <iostream>
+
+#include "common/common/assert.h"
+#include "common/http/http1/parser.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+namespace {
+ParserStatus intToStatus(int rc) {
+  // See
+  // https://github.com/nodejs/llhttp/blob/a620012f3fd1b64ace16d31c52cd57b97ee7174c/src/native/api.h#L29-L36
+  switch (rc) {
+  case HPE_USER:
+    return ParserStatus::Error;
+  case HPE_OK:
+    return ParserStatus::Success;
+  case 1:
+    return ParserStatus::NoBody;
+  case 2:
+    return ParserStatus::NoBodyData;
+  case HPE_PAUSED:
+    return ParserStatus::Paused;
+  default:
+    return ParserStatus::Unknown;
```

Comment:
Coverage report claims that some of these are not covered by tests:
- ParserStatus::Error
- ParserStatus::NoBody
- ParserStatus::NoBodyData
- ParserStatus::Unknown

Reply:
The codec only expects success or paused, and otherwise something went wrong (assert checks). I'll change getStatus to isSuccess or isPaused